### PR TITLE
drop underscore, slight tidy-up

### PR DESCRIPTION
**Pull request by jokeyrhyme**
Firstly, thanks for providing this module, as it's exactly what I've been looking for.

I noticed that it depends on [Underscore.js](http://underscorejs.org/), but only uses it for either `_.extend` or tasks that aren't terrible in vanilla JavaScript. This PR replaces Underscore.js with the [extend-object](https://www.npmjs.com/package/extend-object) module.

I've also:
- started some ESLint options for tooling convenience
- swapped `typeof(value)` usage for `typeof value` for correctness
- updated the README.md test instructions to mention that manually pulling the Ubuntu image is required

I can break this up into separate PRs, or squash it into a single commit, or remove things you don't agree with. `npm test` passes. Let me know what you think. :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
   ],
   "dependencies": {
     "docker-modem": "0.1.x",
-    "underscore": "1.5.x"
+    "extend-object": "^1.0.0"
```
Comment (defunctzombie) [inline package.json]: better to be more exact about dependencies to avoid breaking changes

**Comment by defunctzombie** [comment]
+1

**Comment by apocas** [comment]
Nice! I also wanted to drop underscore in the next release.

Not sure about using "extend-object", since this is a critical but at the same time simple functionality I would think that including a extend function in the util may be a better idea and safer in the long term.

**Comment by jokeyrhyme** [comment]
I made the extend-object dependency more specific for safety.
Did you want me to drop it and hard-code a copy of it or something? Or is that a future discussion?

**Comment by apocas** [comment]
Yeah, I think it's safer to bring the extend functionality into dockerode.
https://github.com/apocas/dockerode/blob/master/lib/util.js this seems a good spot to put it :)

Thanks!

**Comment by jokeyrhyme** [comment]
Done! Want me to squash?

**Comment by apocas** [comment]
Nah not needed.

Thanks!

Later tonight I will publish a new version, where this will also be shipped :)

**Comment by jokeyrhyme** [comment]
Now we just need to get rid of Underscore in docker-modem. :)

**Comment by apocas** [comment]
docker-modem does not use it :)

**Comment by jokeyrhyme** [comment]
Right you are. I misread `npm ls` and thought they both used it, for some reason.

**Comment by apocas** [comment]
Published on v2.1.0

**Pull request merged**